PyInstaller/building/build_main.py

```diff
@@ -570,6 +572,46 @@ def _check_python_library(self, binaries):
         binaries.append((os.path.basename(python_lib), python_lib, 'BINARY'))
         logger.info('Using Python library %s', python_lib)
 
+    def add_conda_dlls(self):
+        """
+        Collect Conda binary dependencies for all packages in ``self.pure``.
+
+        Returns immediately if not in a conda environment.
+        """
+        if not compat.is_pure_conda:
+            return
+        from PyInstaller.utils.hooks import conda
+
+        # Work out which distributions are needed. As distributions can contain
+        # multiple packages, group by distribution to avoid large amounts of
+        # duplicity in ``self.datas`` and in the logs.
+        distributions = collections.defaultdict(list)
+        for (name, *_) in self.pure:
+            distribution = conda.distributions_by_package.get(name)
+            if distribution is None:
+                continue
+            distributions[distribution].append(name)
+
+        # For every distribution being used:
+        for (distribution, imports) in distributions.items():
+            # Collect its DLLs.
+            logger.info("Collect Conda distribution '%s', required by %s.",
+                        distribution.name, imports)
+            dlls = conda.collect_dynamic_libs(distribution.name,
+                                              excludes=self.excludes)
+
+            # Add the DLLs converting from (source, dest_dir) format to
+            # (dest_name, source, 'DATA') format.
+            for (source, dest) in dlls:
+                # I'm intentionally skipping the DLL dependency scanner (by
+                # adding binary files as datas) for two reasons:
+                #   - Conda packages should have brought everything they use
+                #     with them so anything else detected would be system
+                #     libraries.
+                #   - There are litter files (.pdb, .def, ...) which, if
+                #     scanned, cause errors or uncatchable pop-up windows.
+                self.datas.append((os.path.basename(source), source, "DATA"))
```

Comment:
This puts all DLLs in the package root. In the interest of decluttering the root and possibly to make clearer where DLLs come from, would anyone be opposed to putting them in a dedicated `conda-bin` folder and adding a runtime hook to add said `conda-bin` to `PATH`?

Reply:
Not sure how common conda is on macOS, but due to the library loader path rewriting magic we do on binaries there, this will mean you'll need to keep track of which dynlibs go where and adjust the path in the binaries that are linked against them. How much of a problem would that be?

Reply:
I'd prefer to put all extensions and dlls into subfolder, see #2468. So for this pull-request, I'd not touch this, rather finish this on :-)